connection.go

```diff
@@ -99,23 +99,12 @@ func (mc *mysqlConn) handleParams() (err error) {
 	return
 }
 
-func (mc *mysqlConn) markBadConn(err error) error {
-	if mc == nil {
-		return err
-	}
-	if err != errBadConnNoWrite {
-		return err
-	}
-	return driver.ErrBadConn
-}
```

Comment:
I changed my mind.
`markBadConn()` is allowlist for ErrBadConn.

If writePacket() returned driver.ErrBadConn instead of errBadConnNoWrite, all error check need to suppress ErrBadConn.

So until we stop returning ErrBadConn, markBadConn() should alive.

```go
// current
if err := f1(); err != nil {
  return nil, mc.markBadConn(err)
}
if err := f2(); err != nil {
  return nil, err
}
if err := f3(); err != nil {
  return nil, err
}
...

// if we removed markBadConn()
if err := f1(); err != nil {
  return nil, err
}
if err := f2(); err != nil {
  // not safe to retry
  if err == errBadConnNoWrite {
    err = InvalidConn
  }
  return nil, err
}
if err := f3(); err != nil {
  if err == errBadConnNoWrite {
    err = InvalidConn
  }
  return nil, err
}
...

```